microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import time
+import uuid
+from datetime import datetime
+
+from assertpy.assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.sut_orchestrator import AZURE
+from lisa.sut_orchestrator.azure.common import (
+    AzureNodeSchema,
+    get_compute_client,
+    get_node_context,
+    wait_operation,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="Test for VMSnapshot extension ",
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class BVTExtension(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        creates a restore point collection then a restore point which helps in
+        validating the VMSnapshot extension.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_vmsnapshot_extension(
+        self, log: Logger, node: Node, environment: Environment
+    ) -> None:
+        unique_name = str(uuid.uuid4())
+        node_context = get_node_context(node)
+        resource_group_name = node_context.resource_group_name
+        vm_name = node_context.vm_name
+        node_location = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        location = node_location.location
+        restore_point_collection = "rpc_" + unique_name
+        assert environment.platform
+        platform: AzurePlatform = environment.platform  # type: ignore
+        assert isinstance(platform, AzurePlatform)
+        sub_id = platform.subscription_id
+        # creating restore point collection
+        client = get_compute_client(platform)
+        response = client.restore_point_collections.create_or_update(
```

Comment:
is there option to create restore point collection via async operation like we are doing for restore point?

Reply:
I did not find one.